src/DeviceDataStreamValues.tsx

```diff
@@ -0,0 +1,232 @@
+/*
+   This file is part of Astarte.
+
+   Copyright 2024 SECO Mind Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { Card, Container, Table } from 'react-bootstrap';
+import {
+  AstarteDataTuple,
+  AstarteDatastreamIndividualData,
+  AstarteDatastreamObjectData,
+  AstarteInterface,
+  AstarteInterfaceValues,
+} from 'astarte-client';
+import _ from 'lodash';
+
+import BackButton from './ui/BackButton';
+import { useAstarte } from './AstarteManager';
+import { AlertsBanner, useAlerts } from 'AlertManager';
+import 'react-datepicker/dist/react-datepicker.css';
+import RenderAstarteData from 'components/RenderAstarteData';
+import useFetch from 'hooks/useFetch';
+import FilterData from 'components/FilterData';
+
+function formatAstarteData(data?: AstarteDataTuple): string {
+  if (data == null) {
+    return '';
+  }
+  if (_.isArray(data)) {
+    return JSON.stringify(data);
+  }
+  if (_.isBoolean(data)) {
+    return data ? 'true' : 'false';
+  }
+  if (_.isNumber(data)) {
+    return data.toString();
+  }
+  if (_.isNull(data)) {
+    return '';
+  }
+  return String(data);
+}
+
+interface IndividualDatastreamTableProps {
+  data: AstarteDatastreamIndividualData[];
+}
+
+const IndividualDatastreamTable = ({
+  data,
+}: IndividualDatastreamTableProps): React.ReactElement => {
+  if (data.length === 0) {
+    return <p>Maybe device published data but not in the selected timeframe.</p>;
+  }
```

Comment:
This length check should be done by the parent component, so we can make sure the visualized string is the same for Individual and Aggregated datastreams

Reply:
For the string itself i would use: `No data in the selected timeframe`